content/en/docs/tasks/debug/debug-cluster/monitor-node-health.md

```diff
@@ -100,54 +97,58 @@ This approach only applies to a Node Problem Detector started with `kubectl`.
+  [kmsg](https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json),
+  [kernel](https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor-counter.json),
+  [abrt](https://github.com/kubernetes/node-problem-detector/blob/master/config/abrt-adaptor.json),
+  and [systemd](https://github.com/kubernetes/node-problem-detector/blob/master/config/systemd-monitor-counter.json).
 
-The Kernel Monitor matches kernel issues according to a set of predefined rule list in
-[`config/kernel-monitor.json`](https://github.com/kubernetes/node-problem-detector/blob/v0.1/config/kernel-monitor.json). The rule list is extensible. You can expand the rule list by overwriting the
-configuration.
+- A `SystemStatsMonitor` type of daemon collects various health-related system stats as metrics.
+  You can customize its behavior by updating its
+  [configuration file](https://github.com/kubernetes/node-problem-detector/blob/master/config/system-stats-monitor.json).
 
-### Add new NodeConditions
+- A `CustomPluginMonitor` type of daemon invokes and checks various node problems by running
+  user-defined scripts. You can use different custom plugin monitors to monitor different
+  problems and customize the daemon behavior by updating the
+  [configuration file](https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json).
 
-To support a new `NodeCondition`, create a condition definition within the `conditions` field in
-`config/kernel-monitor.json`, for example:
+- A `HealthChecker` type of daemon checks the health of the kubelet and container runtime on a node.
 
-```json
-{
-  "type": "NodeConditionType",
-  "reason": "CamelCaseDefaultNodeConditionReason",
-  "message": "arbitrary default node condition message"
-}
-```
+### Adding support for other log format {#support-other-log-format}
 
-### Detect new problems
+The system log monitor currently supports file-based logs, journald, and kmsg.
+Additional sources can be added by implementing a new
+[log watcher](https://github.com/kubernetes/node-problem-detector/blob/master/pkg/systemlogmonitor/logwatchers/types/log_watcher.go).
 
-To detect new problems, you can extend the `rules` field in `config/kernel-monitor.json`
-with a new rule definition:
+### Adding custom plugin monitors
 
-```json
-{
-  "type": "temporary/permanent",
-  "condition": "NodeConditionOfPermanentIssue",
-  "reason": "CamelCaseShortReason",
-  "message": "regexp matching the issue in the kernel log"
-}
-```
+You can extend the Node Problem Detector to execute any monitor scripts written in any language by
+developing a custom plugin. The monitor scripts must conform to the plugin protocol in exit code
+and standard output. For more information, please refer to the
+[plugin interface proposal](https://docs.google.com/document/d/1jK_5YloSYtboj-DtfjmYKxfNnUxCAvohLnsH5aGCAYQ/edit#).
```

Comment:
Due to the risk of vandalism, let's not link to that doc. We could ask SIG Node to make a tidied version where people with the link only have read access.

Reply:
Posted a comment to the google docs page asking for alternative links. I'm open to any improvement ideas.